### PR TITLE
WS-016: Added Sighting Controller

### PR DESCRIPTION
**Pull request by CadfanWilliams**
Raising on behalf of @charlottemothersole

### CODE REVIEW

File: src/main/resources/static/css/addSighting.css [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+Darkest-Blue: #062844
+Medium-Blue: #064172
+Darker-Blue: #1D77C3
+Lightest-Blue: #42BFDD
+*/
+
+#add-sighting {
+    display: flex;
+    flex-direction: column;
+}
```
Comment (TimLeach635) [inline src/main/resources/static/css/addSighting.css]: Blank line at end of file
```suggestion
}
```

File: src/main/resources/templates/add_sighting.html [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                <label for="longitude">
+                    Longitude
+                </label>
+                <input type="number" required/>
+                <input type="hidden"/>
+                <input type="submit"/>
+            </form>
+        </div>
+    </body>
+</html>
```
Comment (TimLeach635) [inline src/main/resources/templates/add_sighting.html]: Blank line at end of file
```suggestion
</html>
```

File: src/main/resources/templates/add_sighting.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                </label>
+                <input type="date" required/>
+                <label for="latitude">
+                    Latitude
+                </label>
+                <input type="number" required/>
+                <label for="longitude">
+                    Longitude
+                </label>
+                <input type="number" required/>
```
Comment (TimLeach635) [inline src/main/resources/templates/add_sighting.html]: The label `for` attribute needs to connect to `id` attributes on the inputs
```suggestion
                <label for="date">
                    Date of Sighting
                </label>
                <input id="date" type="date" required/>
                <label for="latitude">
                    Latitude
                </label>
                <input id="latitude" type="number" required/>
                <label for="longitude">
                    Longitude
                </label>
                <input id="longitude" type="number" required/>
```
You might need the `name` attribute too, test this out

Comment (TimLeach635) [inline src/main/resources/templates/add_sighting.html]: Prefer space before closing `/>`
```suggestion
                <input type="date" required />
                <label for="latitude">
                    Latitude
                </label>
                <input type="number" required />
                <label for="longitude">
                    Longitude
                </label>
                <input type="number" required />
```

File: src/main/resources/templates/add_sighting.html [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+                <input type="hidden"/>
+                <input type="submit"/>
```
Comment (TimLeach635) [inline src/main/resources/templates/add_sighting.html]: And here too
```suggestion
                <input type="hidden" />
                <input type="submit" />
```

File: src/main/resources/templates/add_sighting.html [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    <head lang="en"
+    xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout"
+    xmlns:th="https://www.thymeleaf.org">
+    <link th:href="@{/css/addSighting.css}" rel="stylesheet" />
+    <link th:href="@{/css/application.css}" rel="stylesheet" />
+    </head>
+    <body>
+        <div class="content-container">
+            <h1>Add a whale sighting</h1>
+            <p>Please add the details of your sighting below.dfrefgerg</p>
```
Comment (TimLeach635) [inline src/main/resources/templates/add_sighting.html]: Typo?
```suggestion
            <p>Please add the details of your sighting below.</p>
```

File: src/main/resources/templates/add_sighting.html [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+            <form id="add-sighting" action="" method="">
```
Comment (TimLeach635) [inline src/main/resources/templates/add_sighting.html]: Please fill in `action` and `method` with the correct values

Comment (TimLeach635) [inline src/main/resources/templates/add_sighting.html]: I think `add-sighting-form` is a more descriptive (and less ambiguous) ID

File: src/main/resources/templates/add_sighting.html [hunk 7]
```diff
@@ -0,0 +1,31 @@
+<!DOCTYPE html>
+<html>
+    <head lang="en"
+    xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout"
+    xmlns:th="https://www.thymeleaf.org">
```
Comment (TimLeach635) [inline src/main/resources/templates/add_sighting.html]: Indenting
```suggestion
    <head
        lang="en"
        xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout"
        xmlns:th="https://www.thymeleaf.org"
    >
```

File: src/main/resources/templates/add_sighting.html [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    <link th:href="@{/css/addSighting.css}" rel="stylesheet" />
+    <link th:href="@{/css/application.css}" rel="stylesheet" />
```
Comment (TimLeach635) [inline src/main/resources/templates/add_sighting.html]: Should be indented as they are inside the `<head>`
```suggestion
        <link th:href="@{/css/addSighting.css}" rel="stylesheet" />
        <link th:href="@{/css/application.css}" rel="stylesheet" />
```

**Review by TimLeach635 — CHANGES_REQUESTED** [review]

File: src/main/resources/templates/add_sighting.html (hunk 3 above)
Comment (charlottemothersole) [reply src/main/resources/templates/add_sighting.html]: I'll add names now - not done the backend for this yet so cant test it at the moment

File: src/main/resources/templates/add_sighting.html (hunk 6 above)
Comment (charlottemothersole) [reply src/main/resources/templates/add_sighting.html]: Uncertain on the action for now - possibly need the th:action atrttribute instead? But have used # for now.

**Review by TimLeach635 — APPROVED** [review]

**Pull request merged**